WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/AddProductName/AddProductNameWithAIView.swift

```diff
@@ -0,0 +1,225 @@
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureTransparentNavigationBar()
+        navigationItem.leftBarButtonItem = UIBarButtonItem(title: Localization.cancel, style: .plain, target: self, action: #selector(dismissView))
+    }
+
+    @objc
+    private func dismissView() {
+        dismiss(animated: true)
+    }
+}
+
+private extension AddProductNameWithAIHostingController {
+    enum Localization {
+        static let cancel = NSLocalizedString("Cancel", comment: "Button to dismiss Add product name screen")
+    }
+}
+
+struct AddProductNameWithAIView: View {
+    @ObservedObject private var viewModel: AddProductNameWithAIViewModel
+    @ScaledMetric private var scale: CGFloat = 1.0
+    @FocusState private var editorIsFocused: Bool
+
+    init(viewModel: AddProductNameWithAIViewModel) {
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        ScrollView {
+            VStack(alignment: .leading, spacing: 0) {
+                VStack(alignment: .leading, spacing: Layout.titleBlockSpacing) {
+                    // Title label.
+                    Text(Localization.title)
+                        .fontWeight(.bold)
+                        .titleStyle()
+
+                    // Subtitle label.
+                    Text(Localization.subtitle)
+                        .foregroundColor(Color(.secondaryLabel))
+                        .bodyStyle()
+                }
+                .padding(.bottom, Layout.titleBlockBottomPadding)
+
+                VStack(alignment: .leading, spacing: Layout.titleBlockSpacing) {
+                    VStack(alignment: .leading, spacing: Layout.editorBlockSpacing) {
+                        Text(Localization.productName)
+                            .foregroundColor(Color(.label))
+                            .subheadlineStyle()
+
+                        ZStack(alignment: .topLeading) {
+                            VStack(spacing: 0) {
+                                TextEditor(text: $viewModel.productNameContent)
+                                    .bodyStyle()
+                                    .foregroundColor(.secondary)
+                                    .padding(insets: Layout.messageContentInsets)
+                                    .frame(minHeight: Layout.minimumEditorHeight, maxHeight: .infinity)
+                                    .focused($editorIsFocused)
+
+                                Divider()
+                                    .frame(height: Layout.dividerHeight)
+                                    .foregroundColor(Color(.separator))
+
+                                suggestNameField
+                                    .padding(insets: Layout.suggestButtonInsets)
+                            }
+                            .overlay(
+                                RoundedRectangle(cornerRadius: Layout.cornerRadius).stroke(Color(.brand))
```

Comment:
👀 The stroke color should be grey when the text field is not in focus:

<img width="289" alt="Screenshot 2023-09-18 at 16 08 30" src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/428f5cd6-2bf4-44b6-9aef-b58cacf7ecb1">

Reply:
Fixed in https://github.com/woocommerce/woocommerce-ios/pull/10723/commits/039beee8905ac4ae0d28d14d2502d5379bd39cf2